app/scripts/views/settings/emails.js

```diff
@@ -49,19 +60,18 @@ define(function (require, exports, module) {
       context.set({
         buttonClass: this._hasSecondaryEmail() ? 'secondary' : 'primary',
         canChangePrimaryEmail: this._canChangePrimaryEmail(),
+        caption: this.translate(t('A secondary email is an additional address for receiving security notices and confirming new Sync devices')),
```

Comment:
oooh, I think I've answered my own question, you are trying to make the mixin generalizable and passing in these values to do so.

Here's what I'd do instead to keep all the mixin related logic in one place....

I would make [upgrade-session-mixin.js](https://github.com/mozilla/fxa-content-server/pull/5626/files#diff-dc819f256d3474aa38b4a155cfaa0939) return a function that takes `options`. Then the function is invoked, it'd return the actual mixin. 

When you call `Cocktail.mixin` to add the mixin to the view, instead of adding `UpgradeSessionMixin` directly, you'd invoke `UpgradeSessionMixin` as a function, it'd return the actual mixin. Then `UpgradeSessionMixin` would have it's own `setInitialContext` method that adds `caption`, `title`, and `gatedHref`.

```js
Cocktail.mixin(
  View,
  ...
  UpgradeSessionMixin({
    caption: t('A secondary email is an additional address for receiving security notices and confirming new Sync devices'),
    gatedHref: 'settings/emails',
    title: t('Secondary email')
  }),
  ...
);
```

To me, this approach is preferable because it is more obvious what consumers need to do when using the mixin. As it is, you have to sprinkle mixin related code far away from where the mixin is added, which'll make it easy to forget to do things.

Reply:
Following on from this comment, I wonder if the initial view gating logic could be simpler to use as well, instead of each mixin calling `return this.setupSessionGateIfRequired()` in it's `beforeRender` if instead the mixin itself had a `beforeRender` that invoked a continuation function that is configurable via the options block. I'm not sure what this would look like, in reality, but might be something to experiment with.

Reply:
> but might be something to experiment with

Tried experimenting with this and I am not sure if it works much better. Maybe we could revisit when we add another panel.